graphs/eacs/src/test/kotlin/com/anaplan/engineering/azuki/graphs/eacs/BEH6.kt

```diff
@@ -0,0 +1,142 @@
+package com.anaplan.engineering.azuki.graphs.eacs
+
+import com.anaplan.engineering.azuki.core.runner.Eac
+import com.anaplan.engineering.azuki.core.system.BEH
+import com.anaplan.engineering.azuki.graphs.adapter.api.GraphBehaviours
+import com.anaplan.engineering.azuki.graphs.dsl.GraphScenario
+import com.anaplan.engineering.azuki.graphs.graphA
+
+@BEH(GraphBehaviours.HasCycle, GraphBehaviours.CreateDirectedGraph, """
+    Determine if one or more cycles exist in graph
+""")
+class BEH6 : GraphScenario() {
+
+    @Eac("When a directed graph has one cycle, return true")
+    fun hasOneCycleDirected() {
+        given {
+            thereIsADirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "c")
+                edge("d", "a")
+                edge("e", "d")
+                edge("c", "e")
+            }
+        }
+        then {
+            hasCycle(graphA, true)
+        }
+    }
+
+    @Eac("When a directed graph has no cycle, return false")
+    fun hasNoCyclesDirected() {
+        given {
+            thereIsADirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "c")
+                edge("d", "a")
+                edge("e", "d")
+            }
+        }
+        then {
+            hasCycle(graphA, false)
+        }
+    }
+
+    @Eac("When a directed graph has a loop but one edge is in the inverse direction, return false")
```

Comment:
🚫 There is indeed an acceptance criterion here, but this is describing a very specific scenario rather than the generic essence of the requirement. I'll leave it to you to come up with the best way to phrase it, but I would say something along the lines that paths which would be cycles in an undirected graph but where two edges are directed towards the same vertex do not count as cycles.

By the way, you cannot just say "return false" here, because there might be other cycles in the graph.